sktime/transformations/series/window_summarizer.py

```diff
@@ -0,0 +1,385 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Extract features across (shifted) windows from provided series."""
+
+__author__ = ["Daniel Bartling"]
+__all__ = ["WindowSummarizer"]
+
+import pandas as pd
+from joblib import Parallel, delayed
+
+from sktime.transformations.base import BaseTransformer
+
+
+class WindowSummarizer(BaseTransformer):
+    """Transformer for extracting time series features.
+
+    The WindowSummarizer transforms input series to features
+    based on a provided dictionary of window summarizer, window shifts
+    and window lengths.
+
+    Parameters
+    ----------
+    Dictionary with the following arguments:
+    index : str, base string of the derived features
+            final name will also contain
+            window shift and window length.
+    win_summarizer: either str corresponding to implemented pandas function, currently
+            * "sum",
+            * "mean",
+            * "median",
+            * "std",
+            * "var",
+            * "kurt",
+            * "min",
+            * "max",
+            * "corr",
+            * "cov",
+            * "skew",
+            * "sem"
+            or function definition
+    window: list of integers
+        Contains values for window shift and window length.
+    target_cols: list of str,
+        Specifies which columns in y or X to target. If set to None will
+        target first column in X.
```

Comment:
I was wondering about this param what it is actually doing? As `y` is always the target and `X` the exog, why do we need this `taget_cols` param? @danbartl

Reply:
@aiwalter you can target columns to transform in X only, target_cols specifies which columns to transform. If you want to transform y, append it to X and add it to target_cols. target_cols will drop the column, but keep the transforms (like lags etc.) Will change the comment to update that.